### PR TITLE
Update hibernate code samples [HZ-1461]

### PR DESCRIPTION
**Pull request by ldziedziul**
Related to https://hazelcast.atlassian.net/browse/HZ-1461

### CODE REVIEW

File: hazelcast-integration/hibernate-2ndlevel-cache/pom.xml [hunk 1]
```diff
@@ -18,9 +18,9 @@
         <!-- needed for checkstyle/findbugs -->
         <main.basedir>${project.parent.parent.basedir}</main.basedir>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <hibernate.version>5.4.10.Final</hibernate.version>
-        <hazelcast.hibernate.version>2.0.0</hazelcast.hibernate.version>
-        <h2db.version>1.4.194</h2db.version>
+        <hibernate.version>5.6.12.Final</hibernate.version>
+        <hazelcast.hibernate.version>2.2.2-SNAPSHOT</hazelcast.hibernate.version>
```
Comment (TomaszGaweda) [inline hazelcast-integration/hibernate-2ndlevel-cache/pom.xml:22]: I don't really like the - SNAPSHOT ending, I guess it's a leftover after working with snapshot version before the release?

File: hazelcast-integration/spring-hibernate-2ndlevel-cache/README.md [hunk 2]
```diff
@@ -29,13 +30,13 @@ In order to configure Hazelcast as second-level cache provider, you need to add
...
 	<artifactId>hazelcast-hibernate53</artifactId>
-	<version>2.0.0</version>
+	<version>2.2.2-SNAPSHOT</version>
 </dependency
 
 <dependency>
 	<groupId>com.hazelcast</groupId>
 	<artifactId>hazelcast</artifactId>
-	<version>4.0.1</version>
+	<version>5.3.0-SNAPSHOT</version>
```
Comment (TomaszGaweda) [inline hazelcast-integration/spring-hibernate-2ndlevel-cache/README.md:39]: 5.2.0 instead?

**Review by TomaszGaweda — CHANGES_REQUESTED** [review]

File: hazelcast-integration/hibernate-2ndlevel-cache/pom.xml (hunk 1 above)
Comment (ldziedziul) [reply hazelcast-integration/hibernate-2ndlevel-cache/pom.xml:22]: In other code samples we use SNAPSHOT versions so I've followed this approach, we encourage users to test new features within code samples ;)

File: hazelcast-integration/spring-hibernate-2ndlevel-cache/README.md (hunk 2 above)
Comment (ldziedziul) [reply hazelcast-integration/spring-hibernate-2ndlevel-cache/README.md]: In other code samples we use SNAPSHOT versions so I've followed this approach, we encourage users to test new features within code samples ;)

File: hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml [hunk 3]
```diff
@@ -28,35 +40,30 @@
...
 			<artifactId>hazelcast-hibernate53</artifactId>
-			<version>2.0.0</version>
+			<version>2.2.2-SNAPSHOT</version>
 		</dependency>
 
 		<dependency>
 			<groupId>com.hazelcast</groupId>
 			<artifactId>hazelcast</artifactId>
-			<version>4.0.1</version>
+			<version>5.3.0-SNAPSHOT</version>
```
Comment (frant-hartm) [inline hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml:54]: We can change this to `${hazelcast.version}` now.

File: hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml [hunk 4]
```diff
@@ -28,35 +40,30 @@
...
-					<artifactId>junit-vintage-engine</artifactId>
-				</exclusion>
-			</exclusions>
 		</dependency>
 
 		<dependency>
 			<groupId>com.hazelcast</groupId>
 			<artifactId>hazelcast-hibernate53</artifactId>
-			<version>2.0.0</version>
+			<version>2.2.2-SNAPSHOT</version>
```
Comment (frant-hartm) [inline hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml:48]: We should update to `2.3.0` when we release.

**Review by frant-hartm — APPROVED** [review]

File: hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml (hunk 4 above)
Comment (ldziedziul) [reply hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml:48]: Yes, will send a PR after the publish

Comment (ldziedziul) [reply hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml]: actually added a commit here: https://github.com/hazelcast/hazelcast-code-samples/pull/550/commits/ec71093961414f289fe2e464dd48059f79b2f92b

File: hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml (hunk 3 above)
Comment (ldziedziul) [reply hazelcast-integration/spring-hibernate-2ndlevel-cache/pom.xml]: fixed: https://github.com/hazelcast/hazelcast-code-samples/pull/550/commits/2d112e9b61c8a3160c611160c8db79d0ab537ba3

**Pull request merged**